src/transports/tcp.rs

```diff
@@ -414,24 +421,28 @@ impl<Id: PeerId> Transport<Id> for TcpTransport<Id> {
         Ok(())
     }
 
-    fn receive(endpoint: &mut Self::Endpoint) -> PeerNetResult<Vec<u8>> {
+    fn receive(
+        endpoint: &mut Self::Endpoint,
+        config: &Self::TransportConfig,
+    ) -> PeerNetResult<Vec<u8>> {
         //TODO: Config one
         let mut len_bytes = vec![0u8; 4];
         endpoint
             .stream
             .read_exact(&mut len_bytes)
             .map_err(|err| TcpError::ConnectionError.wrap().new("recv len", err, None))?;
-        let res_size = u32::from_be_bytes(len_bytes.try_into().map_err(|err| {
+        let res_size = usize::from_be_bytes(len_bytes.try_into().map_err(|err| {
```

Comment:
Can you change this in the `send` also ? :)

Reply:
I'm not sure having `usize` is good when exchanging data over the network as it's not fixed size across architectures.
Having a `u32` or `u64` would be better there IMO

Reply:
switched to u32